### PR TITLE
Initial version of metadata paragraph

### LINKED ISSUE #85: Add or purge metadata?
I still think that we should bring 4th point which would touch on metadata support in DataLad:
- we do have it, even if it has being reworked
  - we do point to other extensions, and I think it would only be beneficial to point to datalad-metalad as "the future"
- it would add value for "DataLad" as addition to git/git-annex
  - flexible metadata extraction and aggregation is a very unique feature
  - it could interest some metadata-savvy folks struggling with their data/metadata archives
- we did get questions/queries about metadata on issues and neurostars, so some users are at least interested in it or do use it
- Figure 1 mentions it ("and metadata", "metadata aggregation") -- if decide to not bother, we should purge from the figure

WDYT?

### PR DESCRIPTION
**Pull request by yarikoptic**
If accepted in some form, Closes #85 

WDYT?

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -245,6 +245,18 @@ the *special remote* protocol provided by git-annex [@git-annex:special_remotes_
...
 
+**Metadata management.**
+Metadata are essential for scientific discovery, as they are routinely used to complete all data analyses.
+Metadata is the core concept behind Git and git-annex functioning:
+Git records and uses metadata about each change (author, date, description, original state, etc) for each commit.
+Git-annex manages metadata about content availability and allows to associate additional arbitrary key-value pairs to any annexed content.
+Files managed by git and git-annex can in turn be of standardized file formats comprised of data with rich metadata records.
+Moreover, entire repositories might conform to a standard (e.g., BIDS [@bids:paper]) or provide a standardized dataset descriptor (e.g., [Frictionless data package](https://frictionlessdata.io/data-package/)).
+To facilitate metadata availability and utility, DataLad provides an extensible framework for metadata extraction and aggregation.
+Metadata for each file (contained in the file or recorded by git and git-annex) or associated with the entire dataset can be extracted into a collection of machine-readable (JSON) records and aggregated across all contained subdatasets.
```
Comment (yarikoptic) [inline paper.md:256]: didn't want to bring up JSON-LD here. May be we should expand metalad description with that aspect?

**Review by bpoldrack — COMMENTED** [review]
I think the metalad description is too weak. It's not just "more efficient and consistent implementation".
For example, it decouples the metadata representation from the data in that it doesn't store the metadata in files in the data branch. We probably don't want to elaborate here before the first release of the new approach, but we shouldn't downplay it at all: It's substantially different and from our point of view it's "the future" of datalad metadata.
Here it sounds like things mostly will stay the way they are.

The message should be (see https://github.com/datalad/datalad-paper-joss/issues/1#issuecomment-802699574):
- metadata is important
- there's a mechanism in datalad core
- we think it should be *replaced* by datalad-metalad

Pointing out the replacement aspect (as in "yeah, we have something, but once it's out, you really should use metalad instead! core is \<some flavor of deprecated\>") is important from my POV.

**Comment by adswa** [comment]
I've read it, but I don't have strong opinions on this paragraph (i.e., completely neutral). I'll leave it to those that do more metadata stuff to apply or propose changes.

**Comment by yarikoptic** [comment]
Re @bpoldrack 's 

> We probably don't want to elaborate here before the first release of the new approach, but we shouldn't downplay it at all: It's substantially different and from our point of view it's "the future" of datalad metadata.

I had no intention to downplay metalad but we also did not want to make metadata too prominent (existing or upcoming implementation).  May be that is what made @adswa to feel just "lukewarm" about this entire paragraph, while I tried to bring the idea that metadata is the core concept behind git and git-annex themselves.  I think the main aspect of importance (original or metalad) is that datalad adds a unique feature of extraction + aggregation of metadata from individual files and entire datasets. It is what neither git nor git-annex provide, and it somewhat follows nicely with prior paragraphs (domain specific, modular).
The major conceptual difference in metadata handling in metalad AFAIK is the technical aspect of storage (no longer in the tree), while also improving interfaces/workflow for "extraction + aggregation" part.  Although the storage aspect is indeed an important one to scale etc, I think it is just a technical aspect of the solution and not worth elaborating on that in this paper, although I guess could be mentioned (please suggest the wording).

**Comment by mih** [comment]
I like it. Just cut the last sentence. There is no need for shame ;-)

File: paper.md [hunk 2]
```diff
@@ -245,6 +245,18 @@ the *special remote* protocol provided by git-annex [@git-annex:special_remotes_
...
+Such simple mechanism makes it possible to provide immediate access to metadata about all contained data within a larger super-dataset (such as [datasets.datalad.org](http://datasets.datalad.org)).
+The currently released implementation of metadata management is being actively reworked in [datalad-metalad](https://github.com/datalad/datalad-metalad/) DataLad extension to provide more efficient and consistent implementation.
```
Comment (yarikoptic) [inline paper.md]:
```suggestion
```

**Comment by yarikoptic** [comment]
ok, cut it. I will merge with that now.

**Pull request merged**